packages/react-ui-components/src/SelectBox/selectBox.js

```diff
@@ -180,6 +180,22 @@ export default class SelectBox extends PureComponent {
         return count;
     }
 
+    handleValueChange = (...rest) => {
+        this.props.onValueChange(...rest);
+        // Clear search box after searching
+        if (this.props.onSearchTermChange) {
+            this.props.onSearchTermChange('');
+        }
+    }
+
+    handleCreateNew = (...rest) => {
+        this.props.onCreateNew(...rest);
+        // Clear search box on creating new
+        if (this.props.onSearchTermChange) {
+            this.props.onSearchTermChange('');
+        }
```

Comment:
Just a little nitpick, but most probably you could create another method that holds this optional reset of the `searchTerm` in one place instead of two :)